### DISCUSSION TITLE
Theia Tree

### DISCUSSION THREAD

**Discussion by ezhilkarthi** [eclipse-theia/theia#9981] [General]
Hi,
I need to load a whole JSON hardcode value in Tree. While Execute dialog, what should be the JSON structure and how should I implement it, is there any example.

**Comment by msujew** [comment]
Hi @ezhilkarthi,

you can always create a [`TreeWidget`](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/tree/tree-widget.tsx) with your custom `TreeModel`. You would need some translation code between your JSON input and the `TreeModel` instance. The `TreeModelImpl` should help you along the way.

**Comment by ezhilkarthi** [comment]
Hi @msujew, How to call the tree widget inside the  ReactComponent.

**Reply by msujew** [reply]
I don't think it's possible to embed the tree widget inside another widget/component. @vince-fugnitto Do you have a better idea?

**Reply by vince-fugnitto** [reply]
> I don't think it's possible to embed the tree widget inside another widget/component. @vince-fugnitto Do you have a better idea?

@msujew I believe it should be possible, if I'm not mistaken the `preferences-widget` is an example of a bigger widget that includes a tree (`PreferencesTreeWidget`):

![image](https://user-images.githubusercontent.com/40359487/131497562-b017ac58-d8a4-45a0-8731-8aa81515e623.png)

https://github.com/eclipse-theia/theia/blob/7442150b335a9280a314dbf30f6a9227305b6b04/packages/preferences/src/browser/views/preference-widget.tsx#L32-L46

https://github.com/eclipse-theia/theia/blob/7442150b335a9280a314dbf30f6a9227305b6b04/packages/preferences/src/browser/views/preference-tree-widget.tsx#L29-L31
